### ISSUE TITLE
FR: Puzzle option: pre-move only

### ISSUE THREAD

**Issue by thisIsTheFoxe** [lichess-org/lichobile#2143]
**Is your feature request related to a problem?**
So, when puzzling I find it easy to just go turn by turn without thinking too much about how the opponent reacts or might react, especially when I'm tired or not that focused. It feels a lot more like a brute force approach where I try all the possibilities until one works and then I keep going with the next move without having/looking for a goal (i.e. checkmate, fork etc.)

**Describe the solution you'd like**
Having an **option** to only allow pre-moves on a puzzle. So, you have to pre-move/solve the entire puzzle with only your pieces until you see the solution. In order to not make it too hard I propose to keep track of the amount of moves (maybe even display it) so that you can't go on forever. If you make as many moves and at least one of them is wrong the puzzle just resets.

**Describe alternatives you've considered**
It's just an idea, lmk what you think or if you think it'd be too hard.

**Additional context**
If that sounds cool, I'd also be interested in working on that at least for the mobile version. :)

**Comment by ErdoganSeref** [comment]
Then you would have to guess which move the other side would play.

**Comment by thisIsTheFoxe** [comment]
@ErdoganSeref yes, that's the idea (and the idea of pre-moves in general).. and as far as I've seen lichess puzzles actually do have the opponent always play the top engine move so it shouldn't be impossible to guess (unlike a certain other chess platform.. 👀)
